lms/lmsweb/views.py

```diff
@@ -83,27 +88,92 @@ def ratelimit_handler(e):
     f'{LIMITS_PER_MINUTE}/minute;{LIMITS_PER_HOUR}/hour',
     deduct_when=lambda response: response.status_code != 200,
 )
-def login(login_error: Optional[str] = None):
+def login(login_message: Optional[str] = None):
 
     if current_user.is_authenticated:
         return get_next_url(request.args.get('next'))
 
     username = request.form.get('username')
     password = request.form.get('password')
     next_page = request.form.get('next')
-    login_error = request.args.get('login_error')
+    login_message = request.args.get('login_message')
     user = User.get_or_none(username=username)
 
     if request.method == 'POST':
-        if user is not None and user.is_password_valid(password):
+        if (
+            user is not None and user.is_password_valid(password)
+            and not user.role.is_not_confirmed
+        ):
             login_user(user)
             return get_next_url(next_page)
-        elif user is None or not user.is_password_valid(password):
-            login_error = 'Invalid username or password'
-            error_details = {'next': next_page, 'login_error': login_error}
+        elif (
+            user is None or not user.is_password_valid(password)
+            or user.role.is_not_confirmed
+        ):
+            login_message = _('שם המשתמש או הסיסמה שהוזנו לא תקינים')
+            if user is not None and user.role.is_not_confirmed:
+                login_message = _('עליך לאשר את המייל')
+            error_details = {'next': next_page, 'login_message': login_message}
             return redirect(url_for('login', **error_details))
 
-    return render_template('login.html', login_error=login_error)
+    return render_template('login.html', login_message=login_message)
+
+
+@webapp.route('/signup', methods=['GET', 'POST'])
+def signup():
+    form = RegisterForm()
+    if form.validate_on_submit():
+        User.get_or_create(**{
+            User.mail_address.name: form.email.data,
+            User.username.name: form.username.data,
+        }, defaults={
+            User.fullname.name: form.fullname.data,
+            User.role.name: Role.get_not_confirmed_role(),
+            User.password.name: form.password.data,
+            User.api_key.name: User.random_password(),
+        })
+
+        send_confirmation_mail(form.email.data, form.fullname.data)
```

Comment:
You don't want to send configuration mail if this user is already exists (+ please add test for this case :pray: )

Reply:
Configuration mail isn't going to be sent because the if condition of the `form.validate_on_submit()` is False if user already exists. Also there is a test for that

Reply:
Ah! Missed that, sorry. So maybe we want to prefer early return there?